addons/stock/tests/test_move.py

```diff
@@ -6379,3 +6379,46 @@ def test_change_move_line_uom(self):
         ml.write({'product_uom_id': self.uom_unit.id})
         self.assertEqual(quant.reserved_quantity, 2)
         self.assertEqual(ml.quantity * self.uom_unit.ratio, 2)
+
+    def test_correct_quantity_autofilled(self):
+        """
+         Check if the quantity is correctly computed when:
+            - The product uom differs from the move uom.
+            - Move lines are manually removed and added back.
+            - The quantity is manually divided into different move lines.
+        """
+        product1 = self.env['product.product'].create({
+            'name': 'product1',
+            'type': 'product',
+            'uom_id': self.env.ref('uom.product_uom_gram').id,
+            'uom_po_id': self.env.ref('uom.product_uom_gram').id
+        })
+        quant = self.env['stock.quant'].create({
+            'product_id': product1.id,
+            'location_id': self.stock_location.id,
+            'quantity': 1000000,
+        })
+        move = self.env['stock.move'].create({
+            'name': 'Test move',
+            'product_id': product1.id,
+            'product_uom_qty': 2,
+            'product_uom': self.env.ref('uom.product_uom_kgm').id,
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+        })
+        move._action_confirm()
+        # remove the exsiting ml
+        move.move_line_ids.unlink()
+        # add a ml
+        line1 = self.env['stock.move.line'].create({
+            'move_id': move.id,
+        })
+        line1.quant_id = quant
```

Comment:
Can't you directly set the value in the above dict, in the SML creation ? 
(Same with L6423)

Reply:
hi @adwid quant_id is a field with store=False
the behaviour is actually different when this val in the "create" and in the "write"